demo/lib/vm.bash

```diff
@@ -69,17 +69,31 @@ vm-check-env() {
         echo "ERROR:"
         return 1
     }
-    if [ ! -e "${HOME}/.ssh/id_rsa.pub" ]; then
+    if [ ! -e "$SSH_KEY".pub ]; then
         echo "ERROR:"
         echo "ERROR: environment check failed:"
-        echo "ERROR:   id_rsa.pub SSH public key not found (but govm needs it)."
+        echo "ERROR:   $SSH_KEY.pub SSH public key not found (but govm needs it)."
         echo "ERROR:"
         echo "ERROR: You can generate it using the following command:"
         echo "ERROR:"
         echo "ERROR:     ssh-keygen"
         echo "ERROR:"
         return 1
     fi
+    if ! ssh-add -l | grep -q "$(SSH_ASKPASS=/bin/false ssh-keygen -l -f "$SSH_KEY" < /dev/null 2>/dev/null)"; then
```

Comment:
before adding, it might be good to ensure that SSH agent is running/set up.
Not all environment have it.

Reply:
I agree, this should probably depend on `[ -n "$SSH_AUTH_SOCK"]`.

Another thing to consider/comment, almost related to this PR: @klihub, @kad, what would you think if we'd reuse existing ssh socket for all ssh connections to the VM? We could store the socket file to govm's working directory for the VM. This speeds up test execution. Once we know the VM name, we could do something like:
```
     echo "# VM SSH server  : ssh $VM_SSH_USER@$VM_IP"

+    if [ -d "$HOME/vms/data/$VM_NAME" ]; then
+        SSH_OPTS="$SSH_OPTS -o ControlMaster=auto -o ControlPath=$HOME/vms/data/$VM_NAME/ssh -o ControlPersist=30"
+        SSH="${SSH%% *} $SSH_OPTS"
+        export SSH SSH_OPTS
+    fi
```

(Or perhaps even better, we could use '%h` in ControlPath so that we would not need to change `SSH` on the fly...)

Reply:
> before adding, it might be good to ensure that SSH agent is running/set up.
> Not all environment have it.

I put in -n + -e $SSH_AGENT check, and added an else branch to complain about encrypted keys
if we have no agent.

Reply:
> Another thing to consider/comment, almost related to this PR: @klihub, @kad, what would you think if we'd reuse existing ssh socket for all ssh connections to the VM? We could store the socket file to govm's working directory for the VM. This speeds up test execution. Once we know the VM name, we could do something like:

I have no experience in using ssh with session multiplexing. All things considered I am a bit doubtful about
any meaningful speed-ups, but if you think it's worth doing then I'm fine with it.